src/main/resources/schema.json

```diff
@@ -32,6 +32,106 @@
             "required": ["testenv"],
             "additionalProperties": false
         },
+        "credentials_string": {
+            "type": "object",
+            "properties": {
+                "type": {
+                    "type": "string",
+                    "enum": ["string", "file", "zip"],
+                    "description": "Fixed ID for the <string>, <file> & <zip> credential type (see Jenkins' Credentials Binding Plugin)"
+                },
+                "id": {
+                    "type": "string",
+                    "description": "Identifier used in Jenkins endpoint for the current credentials"
+                },
+                "variable": {
+                    "type": "string",
+                    "description": "Sets a variable to the text given in the credentials (see Jenkins' Credentials Binding Plugin)"
+                }
+            },
+            "required": ["id"],
+            "additionalProperties": false
+        },
+        "credentials_certificate": {
+            "type": "object",
+            "properties": {
+                "type": {
+                    "type": "string",
+                    "const": "certificate",
```

Comment:
I have a doubt here related to the type definition. From line 187 where credentials are defined as array, then when we access items how would us access to them and review their type?
example:
credentials.items.each { m ->
  switch (m.type)
    case "string": ... break
    ...
    case "certificate": ... break
    ...

or we need to access const for certificate and others similar?

Reply:
`const` is used here to define the fixed value of the property "type", you cannot access it later on within ConfigParser class since it is just relevant for the validation stage. The example you put there (switch) is the right way to check for each credential type